lib/rules/enzyme-no-bad-features.js

```diff
@@ -0,0 +1,102 @@
+module.exports = {
+  meta: {
+    docs: {
+      description: 'Restricts usage of enzyme to the good parts.',
+      category: 'Best Practises',
+      recommended: true,
+      uri:
+        'https://github.com/Shopify/eslint-plugin-shopify/blob/master/docs/rules/enzyme-no-bad-features.md',
+    },
+    schema: [],
+  },
+
+  create(context) {
+    const enzymeWrappers = [];
+    return {
+      'Program:exit': function() {
+        enzymeWrappers.forEach(checkNodeForInvalidProps);
+      },
+      CallExpression(node) {
+        if (!isEnzymeWrapper(node)) {
+          return;
+        }
+
+        enzymeWrappers.push(node);
```

Comment:
I don’t see the benefit of pushing to an array and doing it all on exit. What is the difference from just doing it at the AST node itself?

Reply:
Never mind I see it now